pkgs/tools/X11/xdg-utils/default.nix

```diff
@@ -36,17 +36,15 @@ stdenv.mkDerivation rec {
     cp ${mimisrc}/xdg-open $out/bin/xdg-open
   '' + ''
     sed  '2s#.#\
-    cut()   { ${coreutils}/bin/cut  "$@"; }\
     sed()   { ${gnused}/bin/sed     "$@"; }\
     grep()  { ${gnugrep}/bin/grep   "$@"; }\
     egrep() { ${gnugrep}/bin/egrep  "$@"; }\
     file()  { ${file}/bin/file      "$@"; }\
     awk()   { ${gawk}/bin/awk       "$@"; }\
-    sort()  { ${coreutils}/bin/sort "$@"; }\
     xset()  { ${xset}/bin/xset      "$@"; }\
     perl()  { PERL5LIB=${perlPath} ${perlPackages.perl}/bin/perl "$@"; }\
     mimetype() { ${perlPackages.FileMimeInfo}/bin/mimetype "$@"; }\
-    PATH=$PATH:'"$out"'/bin\
+    PATH=$PATH:'$out'/bin:${coreutils}/bin\
```

Comment:
It seems this was previously not done to avoid polluting `$PATH` (https://github.com/NixOS/nixpkgs/issues/25511#issuecomment-300818382).

I think your argument that xdg-utils scripts will expect other coreutils commands, like head and basename, to be present is reasonable.

Another approach might be to process those xdg scripts to point to the specific tool implementations they expect (e.g. busybox or coreutils), but given that this derivation already pulls in coreutils it seems reasonable to just leverage those.

Reply:
At least the following executables are used from coreutils: head, wc, tr, touch, test, tail. I think it is fine to pollute the ``$PATH`` with all those binaries or we wrap here for days.

Reply:
Having xdg-utils contaminate the environment is bad because it can be confusing. Imagine opening a nix file in a text editor using `xdg-open` and then testing it in a terminal opened from the text editor. Now, arguably, users will likely have coreutils installed so this is more important when adding other programs to `PATH`, or when exporting other environment variables but we should still keep it in mind. Ideally we would use something like @abathur’s [resholved](https://github.com/abathur/resholve).

Reply:
`resholved` isn't in a production-ready state yet and it's also unvavailable in nixpkgs.
Do you propose to wait until `resholved` is released? This could take months, judging its current progress.

Reply:
I haven't read context here so I'm commenting generally: 

I don't want you to be stuck waiting for resholve to land in Nixpkgs (and hate me if it takes a bit).

But, if resholve does *solve* a problem here without needing any new features, I don't think it would take "months" and I'm open to discussing timelines/priorities. 

It shouldn't be too hard to add resholve temporarily to test it out. It has a good core of functionality, and it works for a living in my own profile and in Graham's. I even considered it ready for inclusion earlier this year, but the override API's grown into a bit of a haphazard kitchen sink that I've decided to fix before calling an initial release.



 It is basically at the top of my "priority" list, but some "urgent" projects are taking all of its sunlight.